guides/advanced-use/debugging.md

```diff
@@ -0,0 +1,80 @@
+For most Ember applications, Ember CLI "just works". Run `ember server` in your Terminal and you get
+a LiveReload development server at `http://localhost:4200`. Run `ember build`, and you get a `dist/`
+directory with compiled assets ready to be deployed to your production server.
+
+But things don't always go smoothly and CLI commands can fail inexplicably with error messages that
+can sometimes be difficult to act on. In these cases, it can be useful to have tools to figure out what
+went wrong.
+
+In most programming languages, the two most useful debugging tools are logging and breakpoints. Let's
+cover both of these tools in the context of Ember CLI.
+
+## Logging
+
+Ember CLI's default output is pretty quiet. Under the hood it uses [the `debug` library][7] to log messages.
+You can see this output by setting the `DEBUG` environment variable to `*` or `ember-cli:*`.
+Typically, environment variables specific to a command are set by prefixing the command. For example:
+
+```bash
+DEBUG=ember-cli:* ember build
+```
+
+If you set the variable to `*`. Consult the `debug` library documentation for additional ways to
+focus on the logs you care about.
+
+## Breakpoints
+
+As a JavaScript programmer or Ember Application author, you may be used to using the `debugger` statement
+to pause execution and enter a REPL where you can inspect the current state of the program. By default,
+when you run a Node.js program such as `ember build`, `debugger` statements are ignored. In order to
+pause execution and enter a REPL as you can in a browser, you can use `node inspect`. For example,
+set a `debugger` at the top of `ember-cli-build.js`, and then run:
+
+```bash
+node inspect ./node_modules/.bin/ember build
+```
+
+You'll first see the build command pause at the start of the program. You can tell it to continue
+by typing `cont` (or simply `c`). Once the program begins executing, it will stop at any `debugger`
+statements it finds. You can then enter a REPL, by typing `repl`.
+
+Read more about [debugging Node.js here][1].
+
+## Linking Addons
+
+While logging and breakpoints are useful, there are additional challenges when the code you're trying
```

Comment:
```suggestion
While logging and breakpoints are useful, there are additional challenges when the code you are trying
```

This suggestion is because generally we try to limit contractions in order to make our docs easier for non-native English speakers.